mongo/bulk_write.go

```diff
@@ -248,7 +248,7 @@ func (bw *bulkWrite) runDelete(ctx context.Context, batch bulkWriteBatch) (opera
 func createDeleteDoc(filter interface{}, collation *options.Collation, hint interface{}, deleteOne bool,
 	registry *bsoncodec.Registry) (bsoncore.Document, error) {
 
-	f, err := transformBsoncoreDocument(registry, filter)
+	f, err := transformBsoncoreDocumentMapAllowed(registry, filter)
```

Comment:
This new function is cleaner for the case where maps are allowed, but I think it adds confusion because for the case where maps aren't allowed, there's no `transformBsoncoreDocumentMapsBanned` function. Furthermore, there's no corresponding set of functions for `transformValue`. I'd recommend either making all of these changes or simply going back to `transformBsoncoreDocument` for both cases.

Reply:
done